src/drivers/gps/gps.cpp

```diff
@@ -381,7 +404,10 @@ GPS::task_main()
 					_report_gps_pos.epv = 10000.0f;
 					_report_gps_pos.fix_type = 0;
 
-					if (!(_pub_blocked)) {
+					/* update mavlink hil gps parameter */
+					update_param_hil_gps();
```

Comment:
This leads to a param_find() call in every iteration. Could you please run the param_find() during the initialisation instead? Its a more expensive call than param_get();